lib/interface-types/src/encoders/wat.rs

```diff
@@ -0,0 +1,716 @@
+//! Writes the AST into a string representing WIT with its textual format.
+//!
+//! # Example
+//!
+//! ```rust
+//! use wasmer_interface_types::{
+//!     ast::*,
+//!     encoders::wat::*,
+//!     interpreter::Instruction,
+//! };
+//!
+//! # fn main() {
+//! let input: String = (&Interfaces {
+//!     exports: vec![
+//!         Export {
+//!             name: "foo",
+//!             input_types: vec![InterfaceType::I32],
+//!             output_types: vec![],
+//!         },
+//!         Export {
+//!             name: "bar",
+//!             input_types: vec![],
+//!             output_types: vec![],
+//!         },
+//!     ],
+//!     types: vec![],
+//!     imports: vec![
+//!         Import {
+//!             namespace: "ns",
+//!             name: "foo",
+//!             input_types: vec![],
+//!             output_types: vec![InterfaceType::I32],
+//!         },
+//!         Import {
+//!             namespace: "ns",
+//!             name: "bar",
+//!             input_types: vec![],
+//!             output_types: vec![],
+//!         },
+//!     ],
+//!     adapters: vec![
+//!         Adapter::Import {
+//!             namespace: "ns",
+//!             name: "foo",
+//!             input_types: vec![InterfaceType::I32],
+//!             output_types: vec![],
+//!             instructions: vec![Instruction::ArgumentGet { index: 42 }],
+//!         },
+//!         Adapter::Export {
+//!             name: "bar",
+//!             input_types: vec![],
+//!             output_types: vec![],
+//!             instructions: vec![Instruction::ArgumentGet { index: 42 }],
+//!         },
+//!     ],
+//!     forwards: vec![Forward { name: "main" }],
+//! })
+//!     .into();
+//! let output = r#";; Interfaces
+//!
+//! ;; Interface, Export foo
+//! (@interface export "foo"
+//!   (param i32))
+//!
+//! ;; Interface, Export bar
+//! (@interface export "bar")
+//!
+//! ;; Interface, Import ns.foo
+//! (@interface func $ns_foo (import "ns" "foo")
+//!   (result i32))
+//!
+//! ;; Interface, Import ns.bar
+//! (@interface func $ns_bar (import "ns" "bar"))
+//!
+//! ;; Interface, Adapter ns.foo
+//! (@interface adapt (import "ns" "foo")
+//!   (param i32)
+//!   arg.get 42)
+//!
+//! ;; Interface, Adapter bar
+//! (@interface adapt (export "bar")
+//!   arg.get 42)
+//!
+//! ;; Interface, Forward main
+//! (@interface forward (export "main"))"#;
+//!
+//! assert_eq!(input, output);
+//! # }
+//! ```
+
+use crate::{
+    ast::{Adapter, Export, Forward, Import, InterfaceType, Interfaces, Type},
+    interpreter::Instruction,
+};
+
+/// Encode an `InterfaceType` into a string.
+impl From<&InterfaceType> for String {
```

Comment:
Consider using the [ToString trait](https://doc.rust-lang.org/std/string/trait.ToString.html) here (both is fine, but this seems to be exactly what `ToString` was created for)

Reply:
TIL about `ToString`, thanks!

Addressed in https://github.com/wasmerio/wasmer/pull/787/commits/13cee90d17620c76512f315a061d6d5442f11bbd.